x-pack/metricbeat/module/azure/client.go

```diff
@@ -170,10 +180,50 @@ func (client *Client) MapMetricByPrimaryAggregation(metrics []insights.MetricDef
 		for _, metricName := range metricGroup {
 			metricNames = append(metricNames, *metricName.Name.Value)
 		}
-		if selectedResourceID == "" {
-			selectedResourceID = *resource.ID
-		}
-		clientMetrics = append(clientMetrics, client.CreateMetric(selectedResourceID, resource, resourceSize, namespace, metricNames, key, dim, timegrain))
+		clientMetrics = append(clientMetrics, client.CreateMetric(resourceId, subResourceId, namespace, metricNames, key, dim, timegrain))
 	}
 	return clientMetrics
 }
+
+func (client *Client) GetResourceForData(resId string) Resource {
```

Comment:
Docstring?

Also I would rename `resId` to `resourceId` as it is in `MapMetricByPrimaryAggregation` for consistency and readability.